src/blueapi/worker/reworker.py

```diff
@@ -83,8 +87,26 @@ def __init__(
         self._stopping = Event()
         self._stopped = Event()
 
-    def submit_task(self, name: str, task: Task) -> None:
-        active_task = ActiveTask(name, task)
+    def clear_task(self, task_id: str) -> None:
+        del self._pending_tasks[task_id]
+
+    def get_pending_tasks(self) -> List[Task]:
+        return [active_task.task for active_task in self._pending_tasks.values()]
+
+    def begin_task(self, task_id: str) -> None:
+        task = self._pending_tasks.get(task_id)
+        if task is not None:
+            self._submit_active_task(task)
+        else:
+            raise KeyError(f"No pending task with ID {task_id}")
+
+    def submit_task(self, task: Task) -> str:
+        task_id: str = str(uuid.uuid4())
+        active_task = ActiveTask(task_id, task)
```

Comment:
`ActiveTask` should probably be called something else if it's now being used for non-active tasks